src/firefox/index.js

```diff
@@ -97,14 +99,34 @@ export async function run(
   {
     fxRunner = defaultFxRunner,
     findRemotePort = defaultRemotePortFinder,
-    firefoxBinary, binaryArgs,
+    firefoxBinary,
+    binaryArgs,
+    extensions,
   }: FirefoxRunOptions = {}
 ): Promise<FirefoxInfo> {
 
   log.debug(`Running Firefox with profile at ${profile.path()}`);
 
   const remotePort = await findRemotePort();
 
+  if (firefoxBinary && firefoxBinary.startsWith('flatpak:')) {
+    const flatpakAppId = firefoxBinary.substring(8);
+    log.debug(`Configuring Firefox with flatpak: appId=${flatpakAppId}`);
+
+    // This should be resolved by the fx-runner.
+    firefoxBinary = 'flatpak';
+    binaryArgs = [
+      'run',
+      `--filesystem=${profile.path()}`,
+      ...extensions.map(({ sourceDir }) => `--filesystem=${sourceDir}`),
+      // We need to share the network namespace because we want to connect to
```

Comment:
If for devtools only, Firefox supports UNIX domain sockets with the `--start-debugger-server` option: https://searchfox.org/mozilla-central/rev/c0bed29d643393af6ebe77aa31455f283f169202/devtools/startup/DevToolsStartup.jsm#955

Could that be used to avoid the need for sharing the same network namespace?

Reply:
Maybe but our code does not handle that so we can maybe support UNIX sockets in a follow-up.